components/esp_modem/test/target_urc/main/urc_test.cpp

```diff
@@ -82,25 +101,88 @@ class DCE : public esp_modem::DCE_T<ESP_AT_Module> {
 
     static constexpr int transfer_completed = 1;
 private:
-    static esp_modem::command_result handle_urc(uint8_t *data, size_t len)
+    static esp_modem::DTE::UrcConsumeInfo handle_enhanced_urc(const esp_modem::DTE::UrcBufferInfo &info)
     {
-        static int start_chunk = 0;
-        static int end_chunk = 0;
-        std::string_view chunk((const char*)data + start_chunk, len - start_chunk);
-        int newline = chunk.find('\n');
-        if (newline == std::string_view::npos) {
-            end_chunk = len;    // careful, this grows buffer usage
-            printf(".");
-            return esp_modem::command_result::TIMEOUT;
+        // Log buffer information for debugging
+        ESP_LOGD(TAG, "URC Buffer Info: total_size=%zu, processed_offset=%zu, new_data_size=%zu, command_active=%s",
+                 info.buffer_total_size, info.processed_offset, info.new_data_size,
+                 info.is_command_active ? "true" : "false");
+
+        // Debug: Show buffer content (first 200 chars)
+        if (info.buffer_total_size > 0) {
+            size_t debug_len = std::min(info.buffer_total_size, (size_t)200);
+            ESP_LOGD(TAG, "Buffer content (first %zu chars): %.*s",
+                     debug_len, (int)debug_len, (const char*)info.buffer_start);
         }
-        printf("%.*s\n", newline, (char*)data + start_chunk);
-        start_chunk = end_chunk;
-        // check for the last one
-        constexpr char last_chunk[] = "Transfer completed";
-        if (memmem(data, len, last_chunk, sizeof(last_chunk) - 1) != nullptr) {
+
+        // Create string view of the entire buffer for processing
+        std::string_view buffer((const char*)info.buffer_start, info.buffer_total_size);
+
+        // First, check if we have the completion message anywhere in the buffer
+        if (buffer.find("Transfer completed") != std::string_view::npos) {
```

Comment:
You could use [contains](https://cppreference.com/w/cpp/string/basic_string_view/contains.html)

Reply:
Not unless we move to C++23, esp-modem is still C++17.